histomics_stream/configure.py

```diff
@@ -69,6 +69,49 @@ def __call__(self, slide):
 
         # Do the work.
         if re.compile(r"\.svs$").search(filename):
+            import large_image
+
+            # read whole-slide image file and create large_image object
+            ts = large_image.open(filename)
+
+            # large_image is willing to interpolate resolutions for us, but we will
+            # instead use the actually stored resolutions.
+            # estimated = np.array([ts.getMagnificationForLevel(i)['magnification'] for i in range(ts.levels)])
+
+            # measure objective of level 0
+            objective = np.float32(ts.getNativeMagnification()["magnification"])
+
+            # Note that if ts.getInternalMetadata()["openslide"] does not exist then we
+            # should have an alternative execution path!!!
+
+            # calculate magnifications of levels
+            openslide_dict = ts.getInternalMetadata()["openslide"]
```

Comment:
This will only be set if large_image is using openslide under the hood.  large_image exposes the number of levels directly in the `getMetadata()` method.

Reply:
`ts.getMetaData()` is telling me that there are 10 available levels: 40x, 20x, ..., 0.078125x (numbered 9, 8, ..., 0).  Given that the `ts.getInternalMetadata()['openslide']["openslide.level-count"]` interface is telling me that there are 4 levels, I am concluding that not all 10 are stored in the file.  At the moment we are not using resampling; is there an openslide-independent way to get the number of stored levels and their shapes?

Reply:
large_image will synthesis layers that don't exist from higher resolution layers.

I think the source-independent way of getting actual levels would be `set(ts.getPreferredLevel(level) for level in range(ts.getMetadata()['levels']))`.

Reply:
Thanks!  I have updated the code in the latest push.